.github/workflows/golangci-lint.yml

```diff
@@ -2,8 +2,14 @@ name: golangci-lint
 on:
   pull_request:
     types: [opened, edited, synchronize, reopened]
+permissions:
```

Comment:
Why are we specifying permission top-level and in the job? If I understood it correctly top-level applies to the whole file and the one in the job only to the job. But we only have one job in this file (and in others) so this seems to be redundant?

(same for the other files)